app/Models/Course.php

```diff
@@ -18,14 +18,68 @@ class Course extends Model
         'description'
     ];
 
+    public function scopeSearch($query, $data)
+    {
+        if (isset($data['keyword'])) {
+            $query->where('name', 'LIKE', "%". $data['keyword'] ."%")->orWhere('description', 'LIKE', "%". $data['keyword'] ."%");
+        }
+
+        if (isset($data['created_time'])) {
+            $query->orderBy('courses.created_at', $data['created_time']);
+        }
+
+        if (isset($data['teacher_id']) && !empty($data['teacher_id'])) {
+            $query->join('teacher_course', 'courses.id', '=', 'teacher_course.course_id')
+                ->where('user_id', $data['teacher_id']);
+        }
+
+        if (isset($data['learner']) && !empty($data['learner'])) {
+            $query->withCount('users')->orderBy('users_count', $data['learner']);
+        }
+
+        if (isset($data['time']) && !empty($data['time'])) {
+            $query->withCount('lessons')->withSum('lessons', 'times')->orderBy('lessons_sum_times', $data['time']);
```

Comment:
câu query này được hiểu như thế nào?

Reply:
e có sửa lại thành $query->withSum('lessons', 'times')->orderBy('lessons_sum_times', $data['time']);
select `courses`.*, 
(select sum(`lessons`.`times`) from `lessons` where `courses`.`id` = `lessons`.`course_id` and `lessons`.`deleted_at` is null) as `lessons_sum_times` 
from `courses` 
order by `lessons_sum_times` desc

thực hiện nối bảng courses với bảng lessons,
trả về các bản ghi gồm tất cả các trường trong bảng course và trường lessons_sum_times (là tổng times của lessons theo course_id) và
được sắp xếp theo lessons_sum_times